---
title: "thema07"
author: "keimpe dijkstra"
date: "8-3-2022"
output:
  pdf_document:
    fig_caption: true
    number_sections: true
  html_document:
    df_print: paged
---
# Exploratory data analysis
EnhancedVolcano(j147.toptag$table,x = 'logFC', y = 'FDR', lab = rownames(j147.toptag$table), title = "Volcano plot", subtitle = "J147 vs AD")yr)
library(affy)
library(scales)
library(DESeq2)
library(pheatmap)
library(PoiClaClu)
library(ggplot2)
library(edgeR)
library(EnhancedVolcano)
library(gplots)
library(SPIA)
library(dplyr)
```
\tableofcontents

First, we load all our data into r

```{r}
#load in the data using the read.table method
j147 <- read.table("./data/j147.csv", header = T, sep = ",", quote="")
cad31 <- read.table("./data/cad-31.txt", header = T, sep = "\t", fill = T, quote="")

#create a new column for te gene annotation
cad31$Gene <- strsplit(cad31$Annotation.Divergence, "|", 1)

#filter the gene annotation out of the string
counter <- 1
while( counter < length(cad31$Gene)) {
  
  cad31$Gene[counter] <- cad31$Gene[[counter]][1]
  counter <- counter + 1
}

#drop some unnecesary columns
cad31 <- cad31[c(22, 9:21)]

#merge all the data into one dataframe
data <- merge(j147, cad31)

#replace na's with zero's
data[is.na(data)] <- 0

#rename the columns
names(data) <- c(
                paste0('Gene'),
                paste0('AD.old.j147.', 1:3),
                paste0('AD.old.', 1:3),
                paste0('AD.young.', 1:4),
                paste0('AD.cad31.', 1:3),
                paste0('AD.', 1:3),
                paste0('WT.CAD31.', 1:3),
                paste0('WT.', 1:4) 
                )

#TODO: Check row names just to be sure

#set some indices to help future work
AD.old.j147 <- 2:4
AD.old <- 5:7
AD.young <- 8:11
AD.cad31 <- 12:14
AD <- 15:17
WT.CAD31 <- 18:20
WT <- 21:24
```

## visualizations

basic statistics

```{r}
summary(rowSums(data[AD.old.j147])/3)
summary(rowSums(data[AD.old])/3)
summary(rowSums(data[AD.young])/4)
summary(rowSums(data[AD.cad31])/3)
summary(rowSums(data[AD])/3)
summary(rowSums(data[WT.CAD31])/3)
summary(rowSums(data[WT])/4)
```
The min and first quantile all show similar or very similar results. The wildtype mouse shows the most expression with an almost double maximun than its drugged counterpart.

The young mouse and the mouse on CAD-31 have the least sequences read whereas the mouse with AD and on drugs seem to be upregulated.

This might be explained because of regulation but also the testing can have have a influence on the amount of sequences that are read, therefore we need to normalize the data.


Add 1 to the whole dataframe so it can logscale
```{r}
#set the gene column as row names
row.names(data) <- data$Gene
data[1] <- NULL

data <- data + 1
```

Boxplot
```{r}
myColors <- hue_pal()(7)

boxplot(log2(data),las=2, xlab = "sample mice",
        ylab = "log2 raw gene count", cex.axis=0.5,
        col=rep(myColors,c(3,3,4,3,3,3,4)))

```

Density plot
```{r}
## Plot the log2-transformed data with a 0.1 pseudocount
plotDensity(log2(data + 0.1),
            col=rep(myColors,c(3,3,4,3,3,3,4)),
            lty=c(1:ncol(data)), xlab='Log2(count)',
            main='Expression Distribution')
legend('topright', names(data), lty=c(1:ncol(data)),
       col=rep(myColors,c(3,3,4,3,3,3,4)), cex = 0.3)
abline(v=1.5, lwd=1, col='red', lty=2)
```

Barplot sequence depth
```{r}
barplot(colSums(data)/1000000,    col=rep(myColors,c(3,3,4,3,3,3,4)), 
        xlab = "mouse samples", ylab = "number of reads in millions", cex.names= 0.5, las=2)

abline(h = min(colSums((data)/1000000)), col = "red",
      lty=2, lwd=2)
```

normalization
```{r}

(ddsMat <- DESeqDataSetFromMatrix(countData = data,
                                  colData = data.frame(samples = names(data)),
                                  design = ~ 1))

```

```{r}
rld.dds <- vst(ddsMat)

rld <- assay(rld.dds)



```

```{r}
j.av <- mean(rld[,1:10])
c.av <- mean(rld[,11:23])

j.bi <- mean(rld[,1:10]) +qt(c(0.025, 0.975), length(rld[,1:10])-1)*sd(rld[,1:10]/sqrt(length(rld[,1:10])))
 
c.bi <- mean(rld[,11:23]) +qt(c(0.025, 0.975), length(rld[,11:23])-1)*sd(rld[,11:23]/sqrt(length(rld[,11:23])))

#rld[,11:23] <-rld[,11:23]  * (c.av/j.av)

```


```{r}
sampledist <- dist(t(rld), method = "euclidean", diag = T, upper = T)
sampledistmatrix <- as.matrix(sampledist)
```




heatmap(both datasets)
```{r}
ann <- data.frame(group = factor(c(1,1,1,rep(2,each=7),3,3,3,4,4,4,5,5,5,6,6,6,6),
                                labels = c("AD.J147", "AD.CTRL", "AD.CAD31", "AD.CTRL","WT-CAD31", "WT")),
                  drug = factor(c(rep.int(1,3), rep.int(2,7), rep.int(1,3), rep.int(2,3), rep.int(1,3), rep.int(2,4)), labels = c("on drug", "off drug")),
                  study = factor(c(rep.int(1,10), rep.int(2,13)),
                                 labels = c("study j147", "study cad31")))
                  
row.names(ann) <- names(data)

pheatmap(sampledistmatrix, show_colnames = T,
         annotation_col = ann,
         clustering_distance_rows = sampledist,
         clustering_distance_cols = sampledist,
         main = "Euclidean Sample Distances")                  

```


### heatmap j147
```{r}
ann <- data.frame(group = factor(c(1,1,1,rep(2,each=3), rep(3, each=4)), labels = c("AD.old.J147", "AD.old", "AD.young")),
                  drug = factor(c(rep.int(1,3), rep.int(2,7)), labels = c("on drug", "off drug")))

row.names(ann) <- names(data[1:10])

sampledist <- dist(t(rld[,1:10]), method = "euclidean", diag = T, upper = T)
sampledistmatrix <- as.matrix(sampledist)

pheatmap(sampledistmatrix, show_colnames = T,
         annotation_col = ann,
         clustering_distance_rows = sampledist,
         clustering_distance_cols = sampledist,
         main = "Euclidean Sample Distances")   

```


### heatmap cad-31
```{r}
ann <- data.frame(group = factor(c(rep(1,each=3),rep(2,each=3), rep(3, each=3), rep(4,each=4)), labels = c("AD.CAD31", "AD.CTRL","WT-CAD31", "WT")),
                  drug = factor(c(rep.int(1,3), rep.int(2,3), rep.int(1,3), rep.int(2,4)), labels = c("on drug", "off drug")))

row.names(ann) <- names(data[11:23])

sampledist <- dist(t(rld[,11:23]), method = "euclidean", diag = T, upper = T)
sampledistmatrix <- as.matrix(sampledist)

pheatmap(sampledistmatrix, show_colnames = T,
         annotation_col = ann,
         clustering_distance_rows = sampledist,
         clustering_distance_cols = sampledist,
         main = "Euclidean Sample Distances")   


```

### multi dimensional scaling plot 
```{r}
dds <- assay(ddsMat)
poisd <- PoissonDistance( t(dds), type = "deseq")
# Extract the matrix with distances
samplePoisDistMatrix <- as.matrix(poisd$dd)
# Calculate the MDS and get the X- and Y-coordinates
mdsPoisData <- data.frame( cmdscale(samplePoisDistMatrix) )

# And set some better readable names for the columns
names(mdsPoisData) <- c('x_coord', 'y_coord')
groups <- factor(c(1,1,1,rep(2,each=7),3,3,3,4,4,4,5,5,5,6,6,6,6),
                                labels = c("AD.J147", "AD.CTRL", "AD.CAD31", "AD.CTRL","WT-CAD31", "WT"))

coldata <- names(data)

# Create the plot using ggplot
ggplot(mdsPoisData, aes(x_coord, y_coord, color = groups, label = coldata)) + 
  geom_text(size = 4) +
  ggtitle('Multi Dimensional Scaling') +
  labs(x = "Poisson Distance", y = "Poisson Distance") +
  theme_bw()



```

### conclusions EDA
The first thing that pops out when looking at the images created is that there are big differences between the two datasets.
This can be seen in the heatmap containing both studies and in the mds plot a well, This throws our comparison of gene 
expression a bit of, but it shouldn't be to big of a problem because we can just compare the DEG's later on.

A second thing that can be noticed in the heatmap of the j147 is that the age of the mouse seems of high influence on the 
gene expression but in the mds plot there also is a noticable difference between mice on and of the drug.

The cad-31 study seems to produce some more mixed results, altough harder to detect the same pattern as in the j147 seems
to apply.


```{r}
# Perform a naive FPM normalization
# Note: log transformation includes a pseudocount of 1
counts.fpm <- log2( (data / (colSums(data) / 1e6)) + 1 )

```


```{r}
num.below.one <- 0
num.below.two <- 0

means <- rowMeans(counts.fpm)
difference <- c()

num.with.zero <- 0
num.with.one <- 0
num.with.zero.rows <- c()

i <- 1

while (i <= length(means)){
  if(means[i] < 1){
    num.below.one<- num.below.one+ 1
  }
  if(means[i] < 2){
    num.below.two <- num.below.two +1
  }
  j <- 1
  pass <- F
  
  while(j < length(counts.fpm[i,])){
    if(counts.fpm[i,j] < 1 & pass == F){
      num.with.zero <- num.with.zero +1
      pass <- T
      num.with.zero.rows <- append(num.with.zero.rows, i)
    }
    if(counts.fpm[i,j] < 2){
      num.with.one <- num.with.one +1
      j <- length(counts.fpm[i,])
    }
    j <- j+1
    
  }
  
  difference <- append(difference, max(counts.fpm[i,])-min(counts.fpm[i,]))
  
  i <- i+1}

means <- as.data.frame(means)
means <- cbind(means, difference)
means <- means[order(means[1]),]

plot(means$difference,type="h",xlab="number of genes", ylab="row means", col="darkseagreen1")
lines(means$means,type="l" ,col="blue",)
abline(v=num.below.one, col="red", lty=2)
abline(v=num.below.two, col="red", lty=2)
abline(v=num.with.zero, col="gold", lty=2)
abline(v=num.with.one, col="gold", lty=2)
```
The plot above describes the means of the rows in the normalized data. The abline's in red are values where the means are below 1 and two respectively and the yellow abline's rows that contain values below 1 and 2. The small difference between the abline's below 1 suggests that there isn't much of a difference between the values, the bigger gap between the values of two suggest the opposite.

The difference is seen in the light green histogram in the back, it first shows no difference relative to the means value, but as the mean gets bigger the difference also starts to increase and eventually kinda settles.

Therefore i decide to set the threshold of filtering where none of the values in the row is below one since after that threshold there seems to be a incline of the row means.
```{r}

original_length <- nrow(counts.fpm)
counts.fpm <- counts.fpm[-num.with.zero.rows,]
cat(original_length-nrow(counts.fpm), "rows are deleted")
```

## Fold Change


### j47 old vs old
```{r}
two.groups <- as.data.frame(rowMeans(counts.fpm[AD.old.j147]))
two.groups["AD.old1"] <- rowMeans(counts.fpm[AD.old])
names(two.groups)[1] <- "AD.old.j147"

two.groups["difference"] <- two.groups$AD.old1 - two.groups$AD.old.j147

hist(two.groups$difference, breaks=60, main = "histogram of fold change difference")
abline(v=-1, col="red")
abline(v=1, col="red")
```

### AD.cad31 vs AD
```{r}
two.groups <- as.data.frame(rowMeans(counts.fpm[AD.cad31]))
two.groups["AD"] <- rowMeans(counts.fpm[AD])
names(two.groups)[1] <- "AD.cad31"

two.groups["difference"] <- two.groups$AD - two.groups$AD.cad31

hist(two.groups$difference, breaks=60, main = "histogram of fold change difference")
abline(v=-1, col="red")
abline(v=1, col="red")
```

##    Bio conductor
```{r}
group <- factor(c(1,1,1,rep(2,each=3),3,3,3,3,4,4,4,5,5,5,6,6,6,rep(7, each=4)),
                                labels = c("AD.J147", "AD.CTRL.OLD","AD.CTRL.YOUNG", "AD.CAD31", "AD.CTRL","WT-CAD31", "WT"))
(design <- model.matrix( ~ group))

```

```{r}
data <- data -1
```



```{r echo = FALSE}
dq.dataset <- DESeqDataSetFromMatrix(data, as.data.frame(group), ~0+group)


result <- DESeq(dq.dataset)
DESeqResults.J147 <- results(result, contrast = c("group", "AD.CTRL.OLD", "AD.J147"),
               alpha = 0.05)
DESeqResults.CAD31 <- results(result, contrast = c("group", "AD.CTRL", "AD.CAD31"),
               alpha = 0.05)

DESeqResults.CTRL <- results(result, contrast = c("group", "AD.CTRL", "WT"),
               alpha = 0.05)

print(summary(DESeqResults.J147))
print(summary(DESeqResults.CAD31))
print(summary(DESeqResults.CTRL))

```


```{r echo = FALSE}
DESeq2::plotMA(DESeqResults.J147)
DESeq2::plotMA(DESeqResults.CAD31)
DESeq2::plotMA(DESeqResults.CTRL)
```







```{r}
#function
get_deg <- function(data, group, design){
dl <- DGEList(as.matrix(data), lib.size = colSums(data), norm.factors = rep(1,ncol(data)), group = group)

dl <- calcNormFactors(dl)
plotMDS(dl)

ed <- estimateDisp(dl, design = design)
plotBCV(ed)

et <- exactTest(ed)

toptag<- topTags(et,n = 1000, p.value = 0.05)
return(toptag)
}
```


```{r}
# Make the comparison data
J147.data <- data[1:6]
J147.group <-factor(c(1,1,1,rep(2,each=3)),
                                labels = c("AD.J147", "AD.CTRL.OLD"))
J147.design <- model.matrix( ~ J147.group)


cad31.AD.cad31.data <- data[11:16]
cad31.AD.cad31.group <-factor(c(1,1,1,rep(2,each=3)),
                                labels = c("AD.CAD31", "AD"))
cad31.AD.cad31.design <- model.matrix( ~ cad31.AD.cad31.group)


cad31.wt.cad31.data <- data[17:23]
cad31.wt.cad31.group <-factor(c(1,1,1,rep(2,each=4)),
                                labels = c("WT.CAD31", "WT"))
cad31.wt.cad31.design <- model.matrix( ~ cad31.wt.cad31.group)
```

```{r}
#do the analysis
j147.toptag <- get_deg(J147.data, J147.group, J147.design)
cad31.AD.cad31.toptag <- get_deg(cad31.AD.cad31.data, cad31.AD.cad31.group, cad31.AD.cad31.design)
cad31.wt.cad31.toptag <- get_deg(cad31.wt.cad31.data, cad31.wt.cad31.group, cad31.wt.cad31.design)
```


```{r}
EnhancedVolcano(j147.toptag$table,x = 'logFC', y = 'FDR', lab = rownames(j147.toptag$table))
EnhancedVolcano(cad31.wt.cad31.toptag$table,x = 'logFC', y = 'FDR', lab = rownames(cad31.wt.cad31.toptag$table))
EnhancedVolcano(cad31.AD.cad31.toptag$table,x = 'logFC', y = 'FDR', lab = rownames(cad31.AD.cad31.toptag$table))
```


```{r}
# Create a Venn-diagram given just the list of gene-names for both sets

venn(list("j147 vs AD.ctrl" = rownames(j147.toptag),
          "cad31 vs AD.CTRL" = rownames(cad31.AD.cad31.toptag)), )

venn(list("WT vs CAD31.WT" = rownames(cad31.wt.cad31.toptag),
          "cad31 vs AD.CTRL" = rownames(cad31.AD.cad31.toptag)), )

venn(list("WT vs CAD31.WT" = rownames(cad31.wt.cad31.toptag),
          "j147 vs AD.ctrl" = rownames(j147.toptag)), )

```

```{r}
degs <- c(rownames(j147.toptag) , rownames(cad31.wt.cad31.toptag) , rownames(cad31.AD.cad31.toptag))
j147.degs <- rownames(j147.toptag)
cad31.degs <- rownames(cad31.AD.cad31.toptag)

# Write gene names to a file
write.table(degs, file = "edger-deg-names.txt",
            row.names = FALSE, quote = FALSE, col.names = FALSE)

write.table(j147.degs, file = "edger-j147deg-names.txt",
            row.names = FALSE, quote = FALSE, col.names = FALSE)

write.table(cad31.degs, file = "edger-cad31deg-names.txt",
            row.names = FALSE, quote = FALSE, col.names = FALSE)
```


```{r}
j147.entrezid.gene <- read.table("j147_entrezid_gene.txt", header = T, sep = "\t")
j147.entrezid.gene.conversion <- as.data.frame(rownames(j147.entrezid.gene))
j147.entrezid.gene.conversion$entrez <- j147.entrezid.gene$From
colnames(j147.entrezid.gene.conversion)[1] <- "gene_name"

temp <- as.data.frame(rownames(j147.toptag$table))
temp$logFc <- j147.toptag$table$logFC
colnames(temp)[1] <- "gene_name"

j147.entrezid.logFC <- merge(j147.entrezid.gene.conversion, temp, by="gene_name")

j147.de <- c(j147.entrezid.logFC$logFc)
names(j147.de) <- c(j147.entrezid.logFC$entrez)



cad31.entrezid.gene <- read.table("cad31_entrezid_gene.txt", header = T, sep = "\t", quote = "")
cad31.entrezid.gene.conversion <- as.data.frame(cad31.entrezid.gene$From)
cad31.entrezid.gene.conversion$entrez <- cad31.entrezid.gene$To
colnames(cad31.entrezid.gene.conversion)[1] <- "gene_name"

temp <- as.data.frame(rownames(cad31.AD.cad31.toptag$table))
temp$logFc <- cad31.AD.cad31.toptag$table$logFC
colnames(temp)[1] <- "gene_name"

cad31.entrezid.logFC <- merge(cad31.entrezid.gene.conversion, temp, by="gene_name")

cad31.de <- c(cad31.entrezid.logFC$logFc)
names(cad31.de) <- c(cad31.entrezid.logFC$entrez)

```


```{r}
# Process all signaling pathways to see if they are inhibited or activated
j147.spia.result <- spia(de=j147.de, all=j147.entrezid.logFC$entrez, organism="mmu", plots=TRUE)
cad31.spia.result <- spia(de=cad31.de, all=cad31.entrezid.logFC$entrez, organism="mmu", plots=TRUE)
```




